src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -125,20 +127,23 @@ of this software and associated documentation files (the "Software"), to deal
      * @param clientID The client ID for the created OAuth Application.
      * @param clientSecret The client secret for the created GitHub OAuth Application.
      * @param oauthScopes A comma separated list of OAuth Scopes to request access to.
+     * @param redirectUri An optional redirect URI to be used by GitHub.
      */
     @DataBoundConstructor
     public GithubSecurityRealm(String githubWebUri,
             String githubApiUri,
             String clientID,
             String clientSecret,
-            String oauthScopes) {
+            String oauthScopes,
+            String redirectUri) {
```

Comment:
This may cause upgrade issues.  I'm not sure just a hunch.  I'll test and let you know.  From what I recall it's no longer a good coding practice to modify the constructors of configuration classes.  @jenkinsci/code-reviewers keep me honest!

Reply:
If this needs to change, please give guidance on the approach you'd like to see.

Reply:
See [Constructor vs. setters](https://www.jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters). Is `redirectUri` a mandatory or optional parameter? If optional, it should be a `@DataBoundSetter` rather than a `@DataBoundConstructor`.

Reply:
IMO, missing a test that a serialized older version deserializes into newer version with default value. Which it should looking at marshal/unmarshal.

The reason not to change the constructor is to not break deployments where jenkins instance is initialized in groovy hooks. So at least keep the old constructer signature as well.